sql/src/main/java/com/vmware/ddlog/DDlogJooqProvider.java

```diff
@@ -49,26 +58,38 @@ public DDlogJooqProvider(final DDlogAPI dDlogAPI, final List<String> sqlStatemen
 
     @Override
     public MockResult[] execute(final MockExecuteContext ctx) throws SQLException {
-        final MockResult[] mock = new MockResult[1];
-        // The execute context contains SQL string(s), bind values, and other meta-data
-        final String sql = ctx.sql();
+        final String[] batchSql = ctx.batchSQL();
+        final MockResult[] mock = new MockResult[batchSql.length];
+        try {
+            dDlogAPI.transactionStart();
+            for (int i = 0; i < batchSql.length; i++) {
+                mock[i] = execute(batchSql[i]);
+            }
+            dDlogAPI.transactionCommit();
+        } catch (final DDlogException e) {
+            throw new RuntimeException(e);
+        }
+        return mock;
+    }
 
+    private MockResult execute(final String sql) throws SQLException {
         if (sql.toUpperCase().startsWith("SELECT")) {
-            mock[0] = executeSelectStar(sql);
+            return executeSelectStar(sql);
+        } else if (sql.toUpperCase().startsWith("INSERT INTO")) {
+            return executeInsert(sql);
         } else {
             // Exceptions are propagated through the JDBC and jOOQ APIs
             throw new SQLException("Statement not supported: " + sql);
         }
-        return mock;
     }
 
     private MockResult executeSelectStar(final String sql) throws SQLException {
-        final String[] s = sql.toUpperCase().split(" ");
-        if (!(s.length == 4 && s[1].equals("*") && s[2].equals("FROM"))) {
+        final String[] s = sql.split("[ \t\n]+");
```

Comment:
ideally you should use the presto parser for this too.

Reply:
Yeah, I was thinking the same seeing the other issue with primary keys. I'll refactor that in. I hope it isn't too slow.